# Analysis of titles in politics section of nairaland

## Problem statement

There's no problem statement. This is just an exercise

## Methodology

1. Grab a list of links from <nairaland.com>
1. Filter titles on ones where the word 'buhari', 'pmb' occurs. Ignore case.

In [1]:
from pathlib import Path
import pandas as pd

import hack

In [2]:
# find and collect all the generated excel files
excel_files = Path(Path.joinpath(
    Path().resolve(), 'politics-analysis')
        ).glob('*.xlsx') # generator
excel_files = list(excel_files)

frames = [pd.read_excel(f) for f in excel_files] # list comprehension for speed
df = pd.concat(frames, ignore_index=True) # combine the data into a single frame

In [3]:
# Make all the heading lowercase
df.columns = [x.lower() for x in df.columns]

# convert titles to strings
df.title = df.title.astype(str)
df.title = df.title.str.lower()

# Convert the comments and views columns to numbers
df.views = pd.to_numeric(df.views, errors='coerce')
df.comments = pd.to_numeric(df.comments, errors='coerce')
df['comment_view_ratio'] = df['comments'] / df['views']
df.head(10)

poster                                              title  \
0            Seun                 nairaland says no to secessionists   
1        AcuraZDX  col. ibrahim sakaba's wife issues ultimatum to...   
2  PatrickOkunima  metele attack: poor orientation, not lack of w...   
3       Explorers  videos, pictures of killing of soldiers in met...   
4        yns4real  2019: how atiku plans to beat buhari in southwest   
5        hollakay  efcc to amend charges, obtain arrest warrant a...   
6      Griezwoman  wike’s claims of fg’s plot to plant guns in ri...   
7         ManirBK  2019: obasanjo, jonathan, governors storm yola...   
8    NAIJAHITPLAY  sad: group declares 7–day mourning for nigeria...   
9            Mrop  dino melaye prostrates for oyedepo, pictured w...   

                                                link  comments     views  \
0  http://www.nairaland.com/2792995/nairaland-say...      3215  353351.0   
1  http://www.nairaland.com/4866051/col-ibrahim-s...        91   18981.0   
2  http://www.nairaland.com/4866422/metele-attack...        16     268.0   
3  http://www.nairaland.com/4866126/videos-pictur...        76   11406.0   
4  http://www.nairaland.com/4866387/2019-how-atik...        16     299.0   
5  http://www.nairaland.com/4866115/efcc-amend-ch...        78    8250.0   
6  http://www.nairaland.com/4866202/wikes-claims-...        46    4640.0   
7  http://www.nairaland.com/4866355/2019-obasanjo...        71   16853.0   
8  http://www.nairaland.com/4866538/sad-group-dec...         1       3.0   
9  http://www.nairaland.com/4865905/dino-melaye-p...       125   44312.0   

   last commenter   others  comment_view_ratio  
0       adifala12  12:14am            0.009099  
1       otuekong1   2:42pm            0.004794  
2  PatrickOkunima   2:42pm            0.059701  
3      Ikpongiton   2:42pm            0.006663  
4           Rem22   2:42pm            0.053512  
5         busomma   2:42pm            0.009455  
6            gare   2:42pm            0.009914  
7       ericsmith   2:41pm            0.004213  
8    NAIJAHITPLAY   2:40pm            0.333333  
9          Viktom   2:40pm            0.002821

In [4]:
cond1 = df.title.str.contains('buhari')
cond2 = df.title.str.contains('pmb')
cond3 = df.title.str.contains('atiku')
mask = cond1 | cond2 & cond3

In [5]:
df[mask]

poster                                              title  \
4             yns4real  2019: how atiku plans to beat buhari in southwest   
18          Jombojombo  those expecting buhari to change nigeria will ...   
25               dre11  2019: how atiku plans to beat buhari in southwest   
45             myright  is buhari left handed or right handed? i'm rea...   
48     KikBuhari2Daura  soldiers massacre: reno omokri shares buhari's...   
49           Gistpeeps  boko haram: act like a general, not a pastor –...   
53            nimisola  we will not allow evil forces attain political...   
62     BiafransExposed  jubril/buhari calls his wife mrs buhari instea...   
84              Rilwon  i never plotted a coup throughout my military ...   
105           Aneesaaa  will president buhari show up against atiku fo...   
113     lionjungle3000  nnamdi kanu and buhari’s purported death: the ...   
116         Wisdomkosi  man campaigning for buhari attacked in port ha...   
118             naptu2  "nigeria will prevail over boko haram", presid...   
119         ExAngel007  buhari breaks silence on boko haram's massacre...   
130           cogbuagu  explaination required for buhari right and lef...   
141           hollakay  buhari rescued south-west from political obliv...   
142                340  you can ban the press if they will embarrass a...   
145       mikeycharles  it's not compulsory for buhari to rule from ni...   
146        stephanie11  'buhari declined 15 bills in one day' - dino m...   
157        GentleMoney  count down to 2019 - atiku vs buhari - nairala...   
162            OEPHIUS  death of 100 soldiers :president buhari, it's ...   
180          INTROVERT          is president buhari right or left handed?   
183     ImadeUReadThis  buhari/apc vs gej/pdp, who played the religiou...   
186        kryptochats  anywhere you see buhari zoom his ears -  nnamd...   
195              ugo4u  buhari’s govt, worst in nigeria’s history – ag...   
203          Dammylink  those who lost jobs under buhari is equivalent...   
212             ify590         abandoning buhari's agriculture revolution   
217                 54  is buhari the biggest conman in nigeria's hist...   
221             WhyMoD  is buhari really leading the fight against bok...   
229          Ayodele70  aisha buhari slams apc after they affirmed buh...   
...                ...                                                ...   
60816          6Ebisco  2019 presidency: primate ayodele reveals what ...   
60820          Georgry              after buhari, it’s my turn – okorocha   
60821          Georgry              after buhari, it’s my turn – okorocha   
60822        boman2014  i'll succeed buhari - okorocha declares in owerri   
60826       Idonmakeam  you cant give what you don’t have – ebikapade ...   
60831       sowore2019  the hypocrisy of the buharist on nairaland aga...   
60833    danieljoel759  us president donald trump invites president bu...   
60843           naptu2  ezekwesili reacts to buhari's request to world...   
60852        gistyinka  buhari tells theresa may '' politicians are pr...   
60860          5Ebisco  corrupt elements in buhari’s govt orchestrated...   
60862    Reelectbuhari  lecturer renounces nigerian citizenship, urges...   
60864  Fayose4Presdent  buhari will set nigeria 40 yrs backwards if he...   
60869           neezar                  buhari will win the next election   
60872       presidency  president buhari assures nigerians: your confi...   
60877      GodOfAction         reno omokri mocks buhari and his ministers   
60878      stephanie11  tinubu joins buhari in london for a meeting (p...   
60879          PStacks  2019 elections buhari vs buhari,  either way, ...   
60888      Nehemiah459  "how i will defeat buhari in 2019" - omoyele s...   
60898        CHOPUP411  photos of buhari and theresa may at 10 downing...   
60899        myboy2111                buhari meets british pm theresa may   
60900       

df.style.set_properties(**{'text-align': 'left', 'background-color': '#DF691A', 'color' : 'white', 'font-size' : '125%'})